canonicals.js

```diff
@@ -0,0 +1,53 @@
+var APP = APP || {};
+(function(app) {
+
+    var canonicals = {
+
+        closures: [750486],
+        asynchronous: [14220321],
+        inheritance: [11072556, 7486825]
+
+    };
+
+    var ids = Object.keys(canonicals).map(function(category) {
+        return canonicals[category].join(';');
+    });
+
+    app.xhr.fetch(ids.join(';')).then(function success(questions) {
+        questions.items.forEach(function(question) {
+            Object.keys(canonicals).forEach(function(category) {
+                canonicals[category].forEach(function(q, i) {
+                    //they see me nesting...
+                    if (q === question.question_id) {
+                        canonicals[category][i] = question;
```

Comment:
Can do. I was debating with myself if I should be mutating the lists or not and in the end lazyness prevailed. You're right though, it's not that good an idea.